samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -79,11 +98,12 @@ public void connect() throws ZkInterruptedException {
     }
   }
 
-  public static ZkConnection createZkConnection(String zkConnectString, int sessionTimeoutMs) {
-    return new ZkConnection(zkConnectString, sessionTimeoutMs);
+  // reset all zk-session specific state
+  public void unregister() {
+    ephemeralPath = null;
   }
 
-  ZkClient getZkClient() {
+  public ZkClient getZkClient() {
```

Comment:
@sborya I don't see any usages outside of the current package namespace. Am I missing something? Which tests are you referring to?